quinn-proto/src/connection/mod.rs

```diff
@@ -375,73 +365,24 @@ impl Connection {
         this
     }
 
-    /// Set up qlog for this connection
-    #[cfg(feature = "__qlog")]
-    pub fn set_qlog(
-        &mut self,
-        writer: Box<dyn io::Write + Send + Sync>,
-        title: Option<String>,
-        description: Option<String>,
-        now: Instant,
-    ) {
-        let ty = if self.side.is_server() {
-            VantagePointType::Server
-        } else {
-            VantagePointType::Client
-        };
-
-        let level = EventImportance::Core;
-
-        let trace = TraceSeq::new(
-            VantagePoint {
-                name: None,
-                ty,
-                flow: None,
-            },
-            title.clone(),
-            description.clone(),
-            Some(Configuration {
-                time_offset: Some(0.0),
-                original_uris: None,
-            }),
-            None,
-        );
-
-        let mut streamer = QlogStreamer::new(
-            QLOG_VERSION.to_string(),
-            title,
-            description,
-            None,
-            now,
-            trace,
-            level,
-            writer,
-        );
-
-        if let Err(e) = streamer.start_log() {
-            warn!("could not initialize qlog streamer: {e}");
-            return;
-        }
-
-        self.qlog_streamer = Some(streamer);
-    }
-
     /// Emit a `MetricsUpdated` event to the qlog streamer containing only updated values
-    #[cfg(feature = "__qlog")]
+    #[cfg(feature = "qlog")]
```

Comment:
Would it not be nicer to establish a convention of doing the `cfg` *inside* this function, so that all the callsites do not have to worry about this?  I imagine there will be a lot of these functions eventually, and having all of them take 2 lines to call instead of a single function call could become a bit noisy?  I imagine the compiler would strip the unused function and there would be no runtime overhead.

Reply:
I think this is absolutely the right way to go internally.

I'm waffling on whether we should extend that pattern to the public API. Normally I'd say yes, but I worry that it would be a footgun -- users (particularly unfamiliar users in a support situation) might struggle to understand why configuring and enabling qlogs in the API is doing nothing, but if that API is feature-gated than rustc will explain the oversight at compile time.